app/models/oldCharities/UserAnswerTransformer.scala

```diff
@@ -0,0 +1,225 @@
+        nodeBooleanData(__ \ 'whatYourCharityDoes \ 'promotionOfEfficiencyInArmedForcesPoliceFireAndRescueService ,"armedForcesOfTheCrown")
+      other <- nodeBooleanData(__ \ 'whatYourCharityDoes \ 'whatYourCharityDoesOther ,"other")
+    } yield JsArray(Seq(reliefOfPoverty, education, religion, healthOrSavingOfLives, citizenshipOrCommunityDevelopment,
+      artsCultureOrScience, amateurSport, humanRights, environmentalProtection, reliefOfYouthAge, animalWelfare,
+      armedForcesOfTheCrown, other))
+
+    charitablePurposes.flatMap {
+      arr => if (arr.value.exists(_!=JsString(""))) {
+        (
+          (__ \ 'charitableObjectives).json.copyFrom((__ \ 'whatYourCharityDoes \ 'charitableObjectives).json.pick) and
+            (__ \ 'charitablePurposes).json.put(JsArray(arr.value.filter(_!=JsString("")))) and
+            (__ \ 'whatYourCharityDoesOtherReason).json.copyFrom((__ \ 'whatYourCharityDoes \ 'whatYourCharityDoesOtherReason).json.pick) and
+            ((__ \ 'publicBenefits).json.copyFrom((__ \ 'whatYourCharityDoes \ 'charityThingsBenefitThePublic).json.pick) orElse doNothing) and
+            (__ \ 'isSection4Completed).json.put(JsBoolean(false))
+          ).reduce
+      } else {
+        doNothing
+      }
+    }
+  }
+
+  def toUserAnswersOperationAndFunds : Reads[JsObject] = {
+
+    val futureFunds = for {
+      donations <- nodeBooleanData(__ \ 'operationAndFunds \ 'futureFunds \ 'donations, "donations")
+      fundraising <- nodeBooleanData(__ \ 'operationAndFunds \ 'futureFunds \ 'fundraising, "fundraising")
+      grants <- nodeBooleanData(__ \ 'operationAndFunds \ 'futureFunds \ 'grants, "grants")
+      membershipSubscriptions <- nodeBooleanData(__ \ 'operationAndFunds \ 'futureFunds \ 'membershipSubscriptions, "membershipSubscriptions")
+      tradingIncome <- nodeBooleanData(__ \ 'operationAndFunds \ 'futureFunds \ 'tradingIncome, "tradingIncome")
+      tradingSubsidiaries <- nodeBooleanData(__ \ 'operationAndFunds \ 'futureFunds \ 'tradingSubsidiaries, "tradingSubsidiaries")
+      investmentIncome <- nodeBooleanData(__ \ 'operationAndFunds \ 'futureFunds \ 'investmentIncome, "investmentIncome")
+      other <- nodeBooleanData(__ \ 'operationAndFunds \ 'futureFunds \ 'other, "other")
+    } yield JsArray(Seq(donations, fundraising, grants, membershipSubscriptions, tradingIncome,
+        tradingSubsidiaries, investmentIncome, other))
+
+    val whereWillCharityOperate = for {
+      ccew <- nodeBooleanData(__ \ 'operationAndFunds \ 'whereWillCharityOperate \ 'englandAndWales, "1")
+      oscrv <- nodeBooleanData(__ \ 'operationAndFunds \ 'whereWillCharityOperate \ 'scotland, "2")
+      ccni <- nodeBooleanData(__ \ 'operationAndFunds \ 'whereWillCharityOperate \ 'northernIreland, "3")
+      other <- nodeBooleanData(__ \ 'operationAndFunds  \ 'whereWillCharityOperate\ 'ukWide, "4")
+      overseas <- nodeBooleanData(__ \ 'operationAndFunds  \ 'whereWillCharityOperate\ 'overseas, "5")
```

Comment:
mapping here is changed - in the rewritten service, 1, 2, 3 and 4 stand for, respectively, England, Wales, Scotland and Northern Ireland, but this maps 'old' Scotland to 'new' Wales and 'old' UK-Wide to 'new' Northern Ireland